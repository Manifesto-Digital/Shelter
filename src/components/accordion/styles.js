import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
