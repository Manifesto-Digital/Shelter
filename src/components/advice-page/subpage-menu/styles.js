import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
