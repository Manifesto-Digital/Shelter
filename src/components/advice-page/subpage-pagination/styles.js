import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
