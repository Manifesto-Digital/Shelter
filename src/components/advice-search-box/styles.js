import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
