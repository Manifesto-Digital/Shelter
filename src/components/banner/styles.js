import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
