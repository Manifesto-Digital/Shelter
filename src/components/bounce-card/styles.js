import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
