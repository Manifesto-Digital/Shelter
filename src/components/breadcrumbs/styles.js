import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
