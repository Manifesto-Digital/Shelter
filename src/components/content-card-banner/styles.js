import styled, { css } from 'styled-components';
import { breakpoint } from '../theme/breakpoint';
import { Card } from '../content-card/styles';

export const CardRow = styled.div`
  display: flex;
  flex-flow: column nowrap;
  ${breakpoint.tablet`
    flex-flow: row wrap;
    justify-content: space-between;
  `};

  ${Card} {
    ${breakpoint.tablet`
      ${({ items }) =>
        items &&
        css`
          flex: 0 1 calc(${Math.floor(100 / items)}% - 12px);
          width: calc(${Math.floor(100 / items)}% - 12px);
        `}
    `};
  }
`;

export const BannerBackground = styled.section`
  background: ${props =>
    (props.bannerColour === 'white' && props.theme.palette.white) ||
    (props.bannerColour === 'grey' && props.theme.palette.grey10)};
  padding: ${props => props.theme.spacing.medium} 0;

  ${CardRow} {
    ${Card} {
      flex: ${props => props.bannerFlow === 'vertical' && '0 1 100%'};
      width: ${props => props.bannerFlow === 'vertical' && '100%'};
      max-width: ${props =>
        props.bannerFlow === 'vertical' ? '600px' : 'auto'};
    }
  }
`;

export const HeaderText = styled.h2`
  margin-bottom: ${props => props.theme.spacing.medium};
  padding-top: ${props => props.theme.spacing.standard};

  a {
    text-decoration: none;
  }
`;
