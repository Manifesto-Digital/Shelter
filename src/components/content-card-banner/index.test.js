import React from 'react';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import { hidePascalCaseWarning } from '../../utils/test-mocks';
import ContentCardBanner from './index';
import { BannerBackground, CardRow } from './styles';
import { createContentCardBanner } from '../../utils/test-factories';
import theme from '../theme/variables';

// Default prop values
it('renders correctly', () => {
  const mockBanner = createContentCardBanner();
  snapshotComponent(<ContentCardBanner data={mockBanner} />);
});

hidePascalCaseWarning();

it('Shows a background colour correctly', () => {
  const mockBanner = createContentCardBanner({ bannerColor: 'Grey' });
  const wrapper = mountWithTheme(<ContentCardBanner data={mockBanner} />);

  expect(wrapper.find(BannerBackground)).toHaveStyleRule(
    'background',
    theme.palette.grey10
  );
});

test('Should pass the correct item count', () => {
  const mockCardData = createContentCardBanner();
  const wrapper = mountWithTheme(<ContentCardBanner data={mockCardData} />);
  expect(wrapper.find(CardRow).props().items).toEqual(2);
});
