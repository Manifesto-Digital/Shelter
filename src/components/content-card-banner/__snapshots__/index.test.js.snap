// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c5 {
  position: relative;
  margin-bottom: 20px;
  background: #fff;
  border-radius: 3px;
  box-shadow: 0 1.5px 4px rgba(0,0,0,0.12),0 1.5px 6px rgba(0,0,0,0.06);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  box-shadow: 0 2.5px 6px rgba(0,0,0,0.16),0 2.5px 6px rgba(0,0,0,0.13);
}

.c6 {
  width: 100%;
  border-radius: 3px 3px 0;
}

.c7 {
  position: relative;
}

.c8 {
  position: absolute;
  top: -20px;
  max-width: 100%;
  margin-bottom: 0;
  padding: 5px 20px;
  background-color: #ea232d;
  color: #fff;
  font-size: 16px;
  font-weight: normal;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c9 {
  display: block;
  padding: 20px;
  color: #4d63b5;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
  opacity: 0;
}

.c1 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

.c0 {
  margin-bottom: 40px;
  background: #e5e5e5;
  padding: 30px 0;
}

.c2 {
  margin-bottom: 30px;
}

.c2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width:40.0625em) {
  .c3 {
    -webkit-flex-flow: row wrap;
    -ms-flex-flow: row wrap;
    flex-flow: row wrap;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:40.0625em) {
  .c3 .c4 {
    -webkit-flex: 0 1 calc(50% - 12px);
    -ms-flex: 0 1 calc(50% - 12px);
    flex: 0 1 calc(50% - 12px);
    width: calc(50% - 12px);
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 "
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Our advice and support services across the UK give people <b>one-to-one</p>",
        }
      }
    />
    <div
      className="c3"
    >
      <div
        className="c4 c5"
      >
        <img
          alt=""
          className="c6"
          src="//images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg?f=center&fit=fill&h=350&w=600"
          srcSet="//images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg?f=center&fit=fill&h=350&w=600 480w,
        //images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg?f=center&fit=fill&h=350&w=600"
        />
        <div
          className="c7"
        >
          <h3
            className="c8"
          >
            Shelter Demo Page
          </h3>
          <Link
            className="c9"
            to="/shelter-demo-page"
          >
            A short description for Callams test page
          </Link>
        </div>
        <Link
          className="c10"
          to="/shelter-demo-page"
        >
          A short description for Callams test page
        </Link>
      </div>
      <div
        className="c4 c5"
      >
        <img
          alt=""
          className="c6"
          src="//images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg?f=center&fit=fill&h=350&w=600"
          srcSet="//images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg?f=center&fit=fill&h=350&w=600 480w,
        //images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg?f=center&fit=fill&h=350&w=600"
        />
        <div
          className="c7"
        >
          <h3
            className="c8"
          >
            Shelter Demo Page
          </h3>
          <Link
            className="c9"
            to="/shelter-demo-page"
          >
            A short description for Callams test page
          </Link>
        </div>
        <Link
          className="c10"
          to="/shelter-demo-page"
        >
          A short description for Callams test page
        </Link>
      </div>
    </div>
  </div>
</section>
`;
